libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2950,6 +2950,28 @@ void GCS_MAVLINK::handle_statustext(const mavlink_message_t &msg) const
 }
 
 
+/*
+  handle logging of named values from mavlink.
+ */
+void GCS_MAVLINK::handle_named_value(const mavlink_message_t &msg) const
+{
+    auto *logger = AP_Logger::get_singleton();
+    if (logger == nullptr) {
+        return;
+    }
+    mavlink_named_value_float_t p;
+    mavlink_msg_named_value_float_decode(&msg, &p);
+    char s[11] {};
+    strncpy(s, p.name, sizeof(s)-1);
```

Comment:
This is a bit odd - our strncpy always null-terminates, so usually this would be `sizeof(s)`  That may cause a buffer overread, of course.  Technically `MIN(sizeof(s), sizeof(p.name))`

Nicer would be to
`char s[MAVLINK_MSG_NAMED_VALUE_FLOAT_FIELD_NAME_LEN+1`

Near enough.